### ISSUE TITLE
Question: how to replace/remove http header in HttpRequest?

### ISSUE THREAD

**Issue by Qudratillo** [activej/activej#99]
I am using AsyncHttpClient. 

1. I would like to reuse HttpRequest and change header values.  Is there any way to do that?
Right now, I can see that we can only add http headers to HttpRequest. 

2. There is no method to clone HttpRequest. Can you add it too?

**Comment by eduard-vasinskyi** [comment]
Hello, @Qudratillo 

[HttpRequest](https://github.com/activej/activej/blob/master/core-http/src/main/java/io/activej/http/HttpRequest.java) is not meant to be reused as it is a stateful component. You should create a new `HttpRequest` for each new `AsyncHttpClient#request` call.

We do not think the deletion of the existing HTTP header should be part of an API for several reasons: 
 - It could falsely hint to the user that the `HttpRequest` could be reused
 - HTTP headers are stored in an optimized header multimap, removing elements from which could potentially cause performance/memory issues
 - RFC allows HTTP messages to have multiple headers of the same name. You would need to specify which exact header you want to delete, thus overcomplicating the API

If you absolutely need to remove a header from `HttpRequest`, you should manually copy the `HttpRequest`, omitting the header that you want to remove.

You may do it manually by constructing a new HttpRequest based on an existing `HttpRequest` and then copying headers and body content. 

Usually, you would use HTTP message cloning in rare scenarios, like when implementing HTTP proxy. So, we do not think it should be part of an API.

If you need to create multiple similar HttpRequests, your best option is to create some kind of `HttpRequest` factory to do so.

**Comment by Qudratillo** [comment]
Hello @eduard-vasinskyi 
Thank you for the answer

I don't implement HTTP Proxy. In my case, apiKey used to secure API is refreshed after some time. So whenever I got 401 response, I just need to create new apiKey and replace old apiKey in the HttpRequest. I did it very easily on Spring. However ActiveJ doesn't let me replace it. Okay I got it, I did copy HttpRequest and did what you said. However I experienced several struggles that I think you need to consider:

```

**Comment by eduard-vasinskyi** [comment]
@Qudratillo
Why don’t you just provide the required _apiKey_ when you create a new `HttpRequest` after receiving a 401 response? An `HttpRequest` should be created based on the business logic of an application. If the previous `AsyncHttpClient#request` returned a 401 response, the next `HttpRequest` should contain a new _apiKey_ value.

**Comment by Qudratillo** [comment]
@eduard-vasinskyi 
Yes, I provide new apiKey after copying HttpRequest. I need to create a new HttpRequest and fill it with old HttpRequest attributes (body, headers, url) because I need to execute same request but with new apiKey.

**Comment by Qudratillo** [comment]
@eduard-vasinskyi 
Here what I want to say is copying url is troublesome. As you can see above there is nothing like just taking Url object from old HttpRequest and giving it to new one.

**Comment by eduard-vasinskyi** [comment]
@Qudratillo 
We will add a method to retrieve a complete URL from `HttpResponse` in ActiveJ version 4.4
